### ISSUE TITLE
not working on Android 9 (release apk)

### ISSUE THREAD

**Issue by webforuse** [getsentry/sentry-react-native#655]
OS:
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

_Platform:_
- [ ]  iOS
- [x]  Android

SDK:
- [x]  `@sentry/react-native`
- [ ]  `react-native-sentry`

`react-native` version: 0.60.4

Init Code:

```
Sentry.init({
  dsn: 'https://...@sentry.io/...'
});
```

---
I have following issue:
not working on Android 9 (simulator or device)

Description goes here ...

**Steps to reproduce:**
- Step
build simple application in Android Studio (release.apk)

**Actual result:**
- Actual
Sentry no catching crush or messages

url to repo (simple app)
https://bitbucket.org/webforuse/sandbox/src/master/

**Comment by MacKentoch** [comment]
Same issue.

My context:

- Crash on Android 9 simulator (signed  release build)
- works without issue on iOS 
- "@sentry/react-native": "^1.0.4"
- "react": "16.8.6",
- "react-native": "0.60.5",

here is my logcat:
```bash
com.facebook.react.bridge.UnexpectedNativeTypeException: Value for id cannot be cast from Double to String
        at com.facebook.react.bridge.ReadableNativeMap.checkInstance(ReadableNativeMap.java:141)
        at com.facebook.react.bridge.ReadableNativeMap.getNullableValue(ReadableNativeMap.java:129)
        at com.facebook.react.bridge.ReadableNativeMap.getString(ReadableNativeMap.java:163)
        at io.sentry.RNSentryModule.getUserBuilder(RNSentryModule.java:313)
        at io.sentry.RNSentryModule.sendEvent(RNSentryModule.java:217)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:158)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
        at android.os.Looper.loop(Looper.java:193)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:232)
        at java.lang.Thread.run(Thread.java:764)
```

**Comment by MacKentoch** [comment]
I find the origin of the crash and a workaround (*at least for my case*).

> The origin is `setUser` method that does not support anything else than `string` for data (my case I just needed `id`).


```js
Sentry.setUser({
   id: SOME_ID, // will crash if id is not a string, ex: a number will crash ‼️
});
```

**Comment by nitaking** [comment]
@MacKentoch
Good work!!

**Comment by vdolbilov** [comment]
Can confirm crash logs are not being sent properly when id in `setUser` is an integer on 1.0.6.

**Comment by nbolender** [comment]
Same here

**Comment by zayalabs** [comment]
I simply created an app from scratch using ```react-native```, set up a new project in sentry and added it to the above app. And added a simple ```hello world``` message in ```App.js``` as described in the documentation. 

When I run the app in an emulator or debug mode I am getting the ```hello world```. The events are getting logged. Once I make a ```release.apk```, it doesn't send events only on ```Android 9```. No errors, no app crashed, just not sending any events. I spent a lot of time to finally find that it's only happening in ```Android-9``` and only in a ```release.apk```
